spyder/plugins/editor.py

```diff
@@ -1214,10 +1214,16 @@ def save_focus_editorstack(self):
                 if win.isAncestorOf(editorstack):
                     self.set_last_focus_editorstack(win, editorstack)
 
-    #------ Handling editorstacks
+    # ------ Handling editorstacks
     def register_editorstack(self, editorstack):
         self.editorstacks.append(editorstack)
         self.register_widget_shortcuts(editorstack)
+        if len(self.editorstacks) > 1 and self.main is not None:
+            # The first editostack is registered automatically with Spyder's
+            # main window through the `register_plugin` method. Only additional
+            # editors added by splitting need to be registered.

```

Comment:
Please add the number of the issue here for reference, and I think this approach is ok :+1: Also, when testing this I realized there is similar issue, but with the symbol search (`Ctrl +Alt + p`), it works with the first panel only too. 

For the symbol search, one option to make it work is to try to get the current editor of the Editor plugin when using `goto_line` in the fileswitcher, I think something like this should work:

Change [the `goto_line` method](https://github.com/spyder-ide/spyder/blob/3.x/spyder/widgets/fileswitcher.py#L561) for:
```python
    def goto_line(self, line_number):
        """Go to specified line number in current active editor."""
        if line_number:
            line_number = int(line_number)
            try:
                editor = self.plugin.get_current_editor()
                editor.go_to_line(min(line_number, editor.get_line_count()))
            except AttributeError:
                pass
```
If you have any question let me know, and thanks for working in this one @jnsebgosselin! :)

Reply:
Thank you very much @dalthviz for reviewing this! I'll do the modifications and fix things for symbol search soon.

Reply:
@dalthviz This was not working when tabs were moved in one of the editor and doing a fileswitch or a symbol search in the other. I had to work a little bit harder to make this work. See my comment below.